internal/common/compilation.bzl

```diff
@@ -292,6 +290,10 @@ def compile_ts(ctx,
   if not is_library:
     files += depset(tsickle_externs)
 
+  transitive_es6_srcs = depset()
+  for dep in ctx.attr.deps:
+    transitive_es6_srcs += getattr(dep.typescript, "transitive_es6_srcs", [])
```

Comment:
I think it would be better to do this with an aspect, maybe we should pair on that?

Reply:
I did it myself! Watching you do it yesterday was all I needed